airbyte/sources/base.py

```diff
@@ -667,13 +720,25 @@ def read(
                 skip_validation=skip_validation,
                 progress_tracker=progress_tracker,
             )
+
+            # Restore the original config if it was modified
+            if original_config is not None:
+                self._config_dict = original_config
```

Comment:
Is it safe to put this into a finally block instead of in each exception handler?